Update the hardening guides for rancher 2.7 (#453)
rancher/rancher-docs: docs/reference-guides/rancher-security/hardening-guides/rke2-self-assessment-guide-with-cis-v1.23-k8s-v1.23.md

---
title: RKE2 Self-Assessment Guide - CIS Benchmark v1.23 - K8s v1.23
---

> #### Overview

Reviewer: ```suggestion
```

This document is a companion to the [RKE2 Hardening Guide](rke2-hardening-guide.md). The hardening guide provides prescriptive guidance for hardening a production installation of Rancher with RKE2 provisioned clusters, and this benchmark guide is meant to help you evaluate the level of security of the hardened cluster against each control in the benchmark.

> This guide corresponds to specific versions of Rancher, CIS Benchmark and Kubernetes:

Reviewer: ```suggestion
This guide corresponds to the following versions of Rancher, CIS Benchmarks, and Kubernetes:
```

| Rancher Version | CIS Benchmark Version | Kubernetes Version |
|-----------------|-----------------------|--------------------|
| Rancher v2.7    | Benchmark v1.23       | Kubernetes v1.23   |

Because Rancher and RKE2 install Kubernetes services as containers, many of the control verification checks in the CIS Kubernetes Benchmark do not apply and will have a result of \`Not Applicable\`. This guide will walk through the various controls and provide updated example commands to audit compliance in Rancher created clusters.

> This document is to be used by Rancher operators, security teams, auditors and decision makers.

Reviewer: ```suggestion
This document is for Rancher operators, security teams, auditors and decision makers.
```

For more detail about each audit, including rationales and remediations for failing tests, you can refer to the corresponding section of the CIS Kubernetes Benchmark v1.23. You can download the benchmark, after creating a free account, in [Center for Internet Security (CIS)](https://www.cisecurity.org/benchmark/kubernetes/).

#### Testing controls methodology

> RKE2 launches control plane components as static pods, managed by the kubelet, and uses containerd as the container runtime. Configuration is defined by arguments passed to the container at the time of initialization or via configuration file.

Reviewer: ```suggestion
RKE2 launches control plane components as static pods, managed by the kubelet, and uses containerd as the container runtime. Configuration is defined by arguments passed to the container during initialization or via configuration files.
```

> Where control audits differ from the original CIS benchmark, the audit commands specific to Rancher are provided for testing. When performing the tests, you will need access to the command line on the hosts of all RKE2 nodes. The commands also make use of the [kubectl](https://kubernetes.io/docs/tasks/tools/) (with a valid configuration file) and [jq](https://stedolan.github.io/jq/) tools, which are required in the testing and evaluation of test results.

Reviewer: ```suggestion
Where control audits differ from the original CIS benchmark, the audit commands specific to Rancher are provided for testing. 

When you perform the tests, you'll need access to the command line on the hosts of all RKE2 nodes. The commands also use the [kubectl](https://kubernetes.io/docs/tasks/tools/) (with a valid configuration file) and [jq](https://stedolan.github.io/jq/) tools, which are required for both testing and the evaluation of test results.
```

:::note

> Only `automated` tests (previously called `scored`) are covered in this guide.

Reviewer: ```suggestion
This guide only covers `automated` (previously called `scored`) tests.
```
